tests/integ/scala/test_datatype_suite.py

```diff
@@ -294,22 +330,30 @@ def test_dtypes(session):
     ]
 
 
+@pytest.mark.skipif(
+    "config.getvalue('local_testing_mode')",
+    reason="FEAT: Cast to StructType not supported",
```

Comment:
StructType https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/latest/api/snowflake.snowpark.types.StructType? Then it feels like a bug since StructType represents a table schema, we shouldn't try to cast a column to a table?

Reply:
I think this is a feature -- for live connection, cast to StructType we call `object_construct` underneath: https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/_internal/type_utils.py#L278-L286

Reply:
I see, so the feature work is to support `object_construct`?

Reply:
this is not a bug, we do not have the logic to handle  exp.to being StructType when handling Cast: https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/mock/_plan.py#L1723.

it should be fairly easy support by using the `object_construct` though, I have logged a JIRA: https://snowflakecomputing.atlassian.net/browse/SNOW-1372813